src/coreclr/tools/Common/TypeSystem/IL/ILImporter.cs

```diff
@@ -301,12 +301,17 @@ private void ImportBasicBlocks()
         }
 
         private void MarkBasicBlock(BasicBlock basicBlock)
+        {
+            MarkBasicBlock(basicBlock, ref _pendingBasicBlocks);
```

Comment:
I'm adding the notion of additional lists that we should look at. This helps in the conditional basic block scanning because it lets us defer looking at conditional blocks until after we scanned all the unconditional codepaths. The conditional scanning is rather simplistic and for:

```csharp
static int Method()
{
    if (Foo() == typeof(Bla))
        Expensive();
    return 0;
}
```

We'd see the `Expensive` block is conditioned, then we'd make the `return 0` conditional as well because the control falls through, and then we'd find out that `return 0` is also reachable as a fallthrough without the condition and undo all the conditions (we don't keep track of edges and directions, just the condition).

Looking at conditional blocks last lets us avoid the rollback (see logic in ILImporter.Scanner.cs).